MySQL 8 startup errors

I just tried starting Vitess with MySQL 8.0.12, and got some startup errors. Most of these are `my.cnf` related and not terribly difficult to fix myself. I'm not sure what the long-term strategy is for supporting conflicting config settings. Should I register a new flavor like @dkhenry did in #4193?

```
0 [Warning] [MY-011068] [Server] The syntax 'expire-logs-days' is deprecated and will be removed in a future release. Please use binlog_expire_logs_seconds instead.
0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
0 [System] [MY-013169] [Server] /usr/sbin/mysqld (mysqld 8.0.12) initializing of server in progress as process 18
0 [ERROR] [MY-011071] [Server] unknown variable 'query_cache_size=0'
0 [Warning] [MY-010952] [Server] The privilege system failed to initialize correctly. If you have upgraded your server, make sure you're executing mysql_upgrade to correct the issue.
0 [ERROR] [MY-010119] [Server] Aborting
0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.12)  MySQL Community Server - GPL.
mysqld_safe Logging to '/vtdataroot/tabletdata/error.log'.
mysqld_safe Starting mysqld daemon with databases from /vtdataroot/tabletdata/data
mysqld: Error on realpath() on '/var/lib/mysql-files' (Error 2 - No such file or directory)
0 [Warning] [MY-011068] [Server] The syntax 'expire-logs-days' is deprecated and will be removed in a future release. Please use binlog_expire_logs_seconds instead.
0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
0 [ERROR] [MY-010095] [Server] Failed to access directory for --secure-file-priv. Please make sure that directory exists and is accessible by MySQL Server. Supplied value : /var/lib/mysql-files
0 [ERROR] [MY-010119] [Server] Aborting
mysqld_safe mysqld from pid file /vtdataroot/tabletdata/mysql.pid ended
mysqld_safe Logging to '/vtdataroot/tabletdata/error.log'.
mysqld_safe Starting mysqld daemon with databases from /vtdataroot/tabletdata/data
mysqld: Error on realpath() on '/var/lib/mysql-files' (Error 2 - No such file or directory)
0 [Warning] [MY-011068] [Server] The syntax 'expire-logs-days' is deprecated and will be removed in a future release. Please use binlog_expire_logs_seconds instead.
0 [Warning] [MY-010915] [Server] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
0 [ERROR] [MY-010095] [Server] Failed to access directory for --secure-file-priv. Please make sure that directory exists and is accessible by MySQL Server. Supplied value : /var/lib/mysql-files
0 [ERROR] [MY-010119] [Server] Aborting
mysqld_safe mysqld from pid file /vtdataroot/tabletdata/mysql.pid ende
```

--- a/config/mycnf/default-fast.cnf
+++ b/config/mycnf/default-fast.cnf
@@ -32,8 +32,6 @@ max_connections = 100
 net_write_timeout = 60
 pid-file = {{.PidFile}}
 port = {{.MysqlPort}}
-query_cache_size = 0
-query_cache_type = 0
 # all db instances should start in read-only mode - once the db is started and
 # fully functional, we'll push it into read-write mode
 read-only

--- a/config/mycnf/default-fast.cnf
+++ b/config/mycnf/default-fast.cnf
@@ -37,6 +37,7 @@ port = {{.MysqlPort}}
 read-only
 read_buffer_size = 1M
 read_rnd_buffer_size = 1M
+secure_file_priv = NULL
 server-id = {{.ServerID}}
 skip-name-resolve
 # we now need networking for replication. this is a tombstone to simpler times.

--- a/config/mycnf/default.cnf
+++ b/config/mycnf/default.cnf
@@ -36,6 +36,7 @@ port = {{.MysqlPort}}
 read-only
 read_buffer_size = 1M
 read_rnd_buffer_size = 1M
+secure_file_priv = NULL
 server-id = {{.ServerID}}
 skip-name-resolve
 # all db instances should skip the slave startup - that way we can do any

--- a/config/mycnf/default.cnf
+++ b/config/mycnf/default.cnf
@@ -31,8 +31,6 @@ max_connections = 100
 net_write_timeout = 60
 pid-file = {{.PidFile}}
 port = {{.MysqlPort}}
-query_cache_size = 0
-query_cache_type = 0
 # all db instances should start in read-only mode - once the db is started and
 # fully functional, we'll push it into read-write mode
 read-only

